### PR TITLE
Improve of Python 2 & 3 compatibility

### PR DESCRIPTION
**Pull request by hosekadam**
Mainly in unit test there was used a condition for importing the corresponding variant of module for python 2 or 3, if they have different names. The better way is to use the `six` library.

Changed the conditions to imports through `six`, removed partial compatibility with Python 2.6 (just in some of unit test, not in all of them - module `unittest2`), removed unused imports in tests and small improvement of `utils.py`: changed deprecated `warn` to `warning` which caused deprecated error in unit tests.

RHELC-588

### CODE REVIEW

**Comment by hosekadam** [comment]
I see that there is a new PR496 with converting the tests from unittest to pytest, so maybe this PR can wait until the PR496 would be merged and then rebase this one and apply possible changes.

File: convert2rhel/systeminfo.py
Comment (abadger) [inline convert2rhel/systeminfo.py:30]: I believe that the configparser move is built into six.  So this line is probably not necessary?

**Comment by hosekadam** [comment]
@abadger I've tried to run the tests on CentOS 6 (which is the system with Python 2.6.6) and I got 4 (like unsupported comma etc.) syntax errors, so I think the tests aren't compatible anymore... 

I wanted to try it once again, but unfortunately my terminal crashed, so I (and there is the problem) rebuilt the images what always crashes due `No matching distribution found for pip<10`. Now I think, that I had some leftover container with Centos6, where I was able to run the first attempt of unit tests. Because of that I don't have any output to attach :(

**Comment by hosekadam** [comment]
I'll investigate how to run the unittest under centos 6 again.

**Comment by hosekadam** [comment]
There is running Centos 6 container with all required dependencies. With `make tests6` you can test it. Now I need to fix the unit tests for Python 2.6.6

**Review by abadger — APPROVED** [review]
This looks good to me.  Let's do the python-2.6 fixes in a separate PR. that way we have a checkpoint with all of the improvements that you have here.

**Comment by abadger** [comment]
The coverage.io report that test coverage has decreased with this change is a little strange to me.  From what I can see, the lines it has marked as newly uncovered should not have changed and I do see unittests that should be testing those lines.

**Comment by abadger** [comment]
Running `pytest` with `coverage` On a local centos7 vm with package versions as close to what I see from the coverage github logs I've found a discrepancy in what is being tested:
Local output from pytest --cov convert2rhel
> Name                            Stmts   Miss Branch BrPart  Cover   Missing
> convert2rhel/subscription.py      340     34    148      7    89%   369-371, 391->400, 474-475, 479-480, 508, 518-521, 527-533, 542, 557-558, 565-566, 589-604, 657-658, 706->exit

Output in the github action coverage-centos7 logs:
> Name                            Stmts   Miss Branch BrPart  Cover   Missing
> convert2rhel/subscription.py      335     34    146      7    89%   361-363, 383->392, 466-467, 471-472, 500, 510-513, 519-525, 534, 549-550, 557-558, 581-596, 649-650, 698->exit

For subscription.py (and only subscription.py), the total number of statements found in subscription.py is 5 less than when I run pytest locally.  The listed statements missing coverage all seem to be the same except offset by five lines.  I think that the missing statements account for the .06% drop in coverage but I'm not sure why pytest thinks there are 5 less lines in that file whn run as a github action.

File: Makefile
Comment (r0x0d) [reply Makefile:92]: We don't need to do this right now. Just left thing here, so we can fix in another PR, as this one is good to merge!

**Review by r0x0d — APPROVED** [review]

File: Dockerfiles/centos6.Dockerfile
Comment (abadger) [inline Dockerfiles/centos6.Dockerfile:31]: Just add a comment here and in the script about what this is for:

```suggestion
# The SSL implementation that python-2.6 uses is insecure and pip refuses to install
# packages using it. This script will use curl to download the packages securely and
# then pip can be asked to install from the local files
RUN $PREP_PIP_DEPS
```

File: scripts/centos6_pip_prep.sh
Comment (abadger) [inline scripts/centos6_pip_prep.sh:2]:
```suggestion

# The SSL implementation that python-2.6 uses is insecure and pip refuses to install
# packages using it. This script will use curl to download the packages securely and
# then pip can be asked to install from the local files

```

**Comment by abadger** [comment]
@hosekadam Michal wanted to try out the coverage fix so I moved that to another PR.  I'm going to add the suggestions and rebase this PR to latest.  Hope that's okay with you!

File: scripts/centos6_pip_prep.sh
Comment (abadger) [reply scripts/centos6_pip_prep.sh:2]: Add a comment on why the PREP_PIP_DEPS script is needed

**Comment by hosekadam** [comment]
@abadger Yeah, it's fine :) And thanks for adding the suggestions and rebasing!

File: Makefile
Comment (r0x0d) [inline Makefile:80]: We should add the `tests6` here as well, so when we execute the whole `make tests` command, we get everything executed. 

But this is not a blocker for me. We can merge this and add just this piece of code in another PR.

**Review by r0x0d — APPROVED** [review]
Looks super nice to me. 
I just made one small comment that we can handle later in another PR, no need to add any extra work into this for my comment. 

Also, loved the simplicity of using `six` for both python 2 & 3. Good work!

File: Makefile
Comment (r0x0d) [reply Makefile:80]: Thanks for the clarification!

**Comment by abadger** [comment]
Thanks Adam for all your work on this!

**Pull request merged**
